File: WED/binary_chop_dailies/BinaryChop1.java
public class BinaryChop1 {
  public static int immaSearch(int[] arr, int target, int left, int right) {
    if (left > right) {
      return -1;
    }
    int mid_point = left + (right- left) / 2;

    if (arr[mid_point]==target) {
      return mid_point;
    }
    if (arr[mid_point] > target) {
      return immaSearch(arr, target, left, mid_point - 1);
    }
    return immaSearch(arr, target, mid_point + 1, right);
  }
  public static void main(String[] args) {
    int[] arr = {1,3,45,5,6,67,23,6,23,6,34,4,6,4,63,64,2};
    int target = 63; 

    int result = immaSearch(arr,target, 0, arr.length - 1);
    if (result == -1) {
      System.out.println("I can't seeeeeeeeEEEEEEEEEeeeeeeeeEEEEeeeeeee");
    } else {
      System.out.println("I found!!! at "+result);
    }
  }
}

File: WED/binary_chop_dailies/binary_chop1.rb
def chop(target, array)
  left = 0
  right = array.length - 1
  while left <= right
    mid = (left + right) / 2
    guess = array[mid]
    if guess == target
      return mid
    elsif guess < target
      left = mid + 1
    else
      right = mid - 1
    end
  end
  -1
end

def verify(i32)
  i32 > 0 ? "Found!" : "Can't find em"
end

puts "First Experiment: target: 2"
puts chop(2, [1,3,4,5,6])
puts verify(chop(2, [1,3,4,5,6]))
puts "Second Experiment: target: 2"
puts verify(chop(2, [1,2,4,5,6]))
puts chop(2, [1,2,4,5,6])
